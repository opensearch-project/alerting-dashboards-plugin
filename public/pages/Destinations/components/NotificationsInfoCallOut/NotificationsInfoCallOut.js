/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiCallOut, EuiButton, EuiLink, EuiSpacer } from '@elastic/eui';
import { MANAGE_CHANNELS_PATH } from '../../../CreateTrigger/utils/constants';
import { NOTIFICATIONS_LEARN_MORE_HREF } from '../../utils/constants';

const noNotificationsTitle = 'Unable to send notifications. Notifications plugin is required.';
const noNotificationsBodyText = (
  <>
    <p>
      Destinations will be deprecated going forward. Install the Notifications plugin for a new
      centralized place to manage your notification channels.
    </p>
    <p>
      Existing destinations will be automatically migrated once Notifications plugin is installed.
    </p>
  </>
);
const noNotificationsButton = (
  <EuiButton
    external
    href={NOTIFICATIONS_LEARN_MORE_HREF}
    iconSide={'right'}
    iconType={'popout'}
    target={'_blank'}
  >
    View install instructions
  </EuiButton>
);

const hasNotificationsTitle = 'Destinations have become channels in Notifications.';
const hasNotificationsBodyText = (
  <p>
    Your destinations have been migrated to Notifications, a new centralized place to manage your
    notification channels. Destinations will be deprecated going forward.&nbsp;
    <EuiLink external href={NOTIFICATIONS_LEARN_MORE_HREF} target={'_blank'}>
      Learn more
    </EuiLink>
  </p>
);
const hasNotificationsButton = (
  <EuiButton href={MANAGE_CHANNELS_PATH}>View in Notifications</EuiButton>
);

const NotificationsInfoCallOut = ({ hasNotificationPlugin }) => {
  return (
    <div>
      <EuiCallOut
        color={hasNotificationPlugin ? 'primary' : 'danger'}
        iconType={hasNotificationPlugin ? undefined : 'alert'}
        title={hasNotificationPlugin ? hasNotificationsTitle : noNotificationsTitle}
      >
        {hasNotificationPlugin ? hasNotificationsBodyText : noNotificationsBodyText}
        <EuiSpacer size={'l'} />
        {hasNotificationPlugin ? hasNotificationsButton : noNotificationsButton}
      </EuiCallOut>
      <EuiSpacer size={'l'} />
    </div>
  );
};

export default NotificationsInfoCallOut;
