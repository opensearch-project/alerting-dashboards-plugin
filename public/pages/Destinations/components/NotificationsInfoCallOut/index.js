/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import NotificationsInfoCallOut from './NotificationsInfoCallOut';

export default NotificationsInfoCallOut;
