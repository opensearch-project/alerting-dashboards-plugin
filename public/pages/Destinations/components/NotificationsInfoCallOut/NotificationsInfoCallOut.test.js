/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import NotificationsInfoCallOut from './NotificationsInfoCallOut';

describe('NotificationsInfoCallOut', () => {
  test('renders when Notifications plugin is installed', () => {
    const component = <NotificationsInfoCallOut hasNotificationPlugin={true} />;

    expect(render(component)).toMatchSnapshot();
  });
  test('renders when Notifications plugin is not installed', () => {
    const component = <NotificationsInfoCallOut hasNotificationPlugin={false} />;

    expect(render(component)).toMatchSnapshot();
  });
});
