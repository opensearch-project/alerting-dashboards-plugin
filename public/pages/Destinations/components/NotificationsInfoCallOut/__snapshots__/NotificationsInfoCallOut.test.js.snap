// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsInfoCallOut renders when Notifications plugin is installed 1`] = `
<div>
  <div
    class="euiCallOut euiCallOut--primary"
  >
    <div
      class="euiCallOutHeader"
    >
      <span
        class="euiCallOutHeader__title"
      >
        Destinations have become channels in Notifications.
      </span>
    </div>
    <div
      class="euiText euiText--small"
    >
      <div
        class="euiTextColor euiTextColor--default"
      >
        <p>
          Your destinations have been migrated to Notifications, a new centralized place to manage your notification channels. Destinations will be deprecated going forward.Â 
          <a
            class="euiLink euiLink--primary"
            href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
        </p>
        <div>
          <a
            class="euiLink euiLink--primary"
            href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
            rel="noopener noreferrer"
            target="_blank"
          >
            EuiIconMock
          </a>
        </div>
        <a
          class="euiLink euiLink--primary"
          href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
        <p />
        <div
          class="euiSpacer euiSpacer--l"
        />
        <a
          class="euiButton euiButton--primary"
          href="/app/notifications-dashboards#/channels"
          rel="noreferrer"
        >
          <span
            class="euiButtonContent euiButton__content"
          >
            <span
              class="euiButton__text"
            >
              View in Notifications
            </span>
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
</div>
`;

exports[`NotificationsInfoCallOut renders when Notifications plugin is not installed 1`] = `
<div>
  <div
    class="euiCallOut euiCallOut--danger"
  >
    <div
      class="euiCallOutHeader"
    >
      <div>
        EuiIconMock
      </div>
      <span
        class="euiCallOutHeader__title"
      >
        Unable to send notifications. Notifications plugin is required.
      </span>
    </div>
    <div
      class="euiText euiText--small"
    >
      <div
        class="euiTextColor euiTextColor--default"
      >
        <p>
          Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
        </p>
        <p>
          Existing destinations will be automatically migrated once Notifications plugin is installed.
        </p>
        <div
          class="euiSpacer euiSpacer--l"
        />
        <a
          class="euiButton euiButton--primary"
          href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            <div>
              EuiIconMock
            </div>
            <span
              class="euiButton__text"
            >
              View install instructions
            </span>
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
</div>
`;
